jenkins/deduper.py

```diff
@@ -137,24 +178,29 @@ def stop_duplicates(self, build_data):
 def deduper_main(raw_args):
     # Get args
     parser = argparse.ArgumentParser(
-        description="Programatically abort oldest builds for"
-                    "a PR so that only one is running.")
-    parser.add_argument('--token', '-t', dest='token',
-                        help='jeknins api token', required=True)
-    parser.add_argument('--user', '-u', dest='username',
-                        help='jenkins username', required=True)
-    parser.add_argument('--job', '-j', dest='job_url',
-                        help='URL of jenkins job that uses the GHPRB plugin',
-                        required=True)
-    parser.add_argument('--log-level', dest='log_level',
-                        default="INFO", help="set logging level")
+        description="Programatically abort older, still-running builds for a PR.")
+    parser.add_argument(
+        '--token', '-t', dest='token',
+        help='jeknins api token', required=True)
+    parser.add_argument(
+        '--user', '-u', dest='username',
+        help='jenkins username', required=True)
+    parser.add_argument(
+        '--job', '-j', dest='job_url', required=True,
+        help='URL of jenkins job that uses the GHPRB plugin')
+    parser.add_argument(
+        '--one-per-author', dest='one_per_author', default=False,
+        help="Optionaly leave one build running for each unique last commit author for a PR.")
+    parser.add_argument(
+        '--log-level', dest='log_level',
+        default="INFO", help="set logging level")
     args = parser.parse_args(raw_args)
 
     # Set logging level
     logging.getLogger().setLevel(args.log_level.upper())
 
     # Abort extra jobs
-    job = JenkinsJob(args.job_url, args.username, args.token)
+    job = JenkinsJob(args.job_url, args.username, args.token, args.one_per_author)
```

Comment:
It would opt to pass `args.one_per_author` directly to deduper instead of to job, since it is only used in the `JenkinsJob` class so that it can be used in `GhprbOutdatedBuildAborter`.